crates/sozo/src/commands/options/account.rs

```diff
@@ -322,4 +324,51 @@ mod tests {
             .is_err()
         );
     }
+
+    #[tokio::test]
+    async fn legacy_flag_works_as_expected() {
+        let cmd = Command::parse_from([
+            "sozo",
+            "--legacy",
+            "--account-address",
+            "0x0",
+            "--private-key",
+            "0x1",
+        ]);
+        let (_runner, provider) = KatanaRunner::new().unwrap();
+        let dummy_call = vec![Call {
+            to: FieldElement::from_hex_be("0x0").unwrap(),
+            selector: FieldElement::from_hex_be("0x1").unwrap(),
+            calldata: vec![
+                FieldElement::from_hex_be("0x2").unwrap(),
+                FieldElement::from_hex_be("0x3").unwrap(),
+            ],
+        }];
+
+        // HACK: SingleOwnerAccount doesn't expose a way to check `encoding` type used in struct, so
+        // checking it by encoding a dummy call and checking which method it used to encode the call
+        let account = cmd.account.account(provider, None).await.unwrap();
+        let result = account.encode_calls(&dummy_call);
+        assert!(*result.get(3).unwrap() == FieldElement::from_hex_be("0x0").unwrap());
+    }
+
+    #[tokio::test]
+    async fn without_legacy_flag_works_as_expected() {
+        let cmd = Command::parse_from(["sozo", "--account-address", "0x0", "--private-key", "0x1"]);
+        let (_runner, provider) = KatanaRunner::new().unwrap();
+        let dummy_call = vec![Call {
+            to: FieldElement::from_hex_be("0x0").unwrap(),
+            selector: FieldElement::from_hex_be("0x1").unwrap(),
+            calldata: vec![
+                FieldElement::from_hex_be("0x2").unwrap(),
+                FieldElement::from_hex_be("0x3").unwrap(),
```

Comment:
```suggestion
                FieldElement::from_hex_be("0xf2").unwrap(),
                FieldElement::from_hex_be("0xf3").unwrap(),
```
To clearly identify which field in the encoded data we're testing here.